root_numpy/src/tree.pyx

```diff
@@ -10,8 +10,14 @@ cdef list_objects_recursive(TDirectory* rdir, objects, types=None, path=""):
     for i in range(nkeys):
         key = <TKey*> keys.At(i)
         clsname = str(key.GetClassName())
-        if types is None or clsname in types:
+        cl = ROOT.gROOT.GetClass(clsname)
+        if types is None:
             objects.append(path + str(key.GetName()))
+        else:
+            for t in types:
+                if cl.InheritsFrom(ROOT.gROOT.GetClass(t)):
+                    objects.append(path + str(key.GetName()))
+                    break
```

Comment:
why not something like:
`if types is None or if numpy.any([cl.InheritsFrom(ROOT.gROOT.GetClass(t)) for t in types])`